pkg/kubelet/network/plugins.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	"github.com/golang/glog"
+)
+
+// Plugin is an interface to network plugins for the kubelet
+type NetworkPlugin interface {
+	// Validate tests the validity of the plugin. The implementation
+	// should check whether any appropriate packages are installed
+	//  correctly or not. etc.
+	Validate() error
+
+	// Init initializes the plugin.  This will be called exactly once
+	// before any other methods are called.
+	Init(host Host) error
+
+	// Name returns the plugin's name. This will be used when searching
+	// for a plugin by name, e.g.
+	Name() string
+
+	// PrePodLaunch is the method called after the infra container of
+	// the pod has been created but before the other containers of the
+	// pod are launched.
+	PrePodLaunch(name string, namespace string, podInfraContainerID dockertools.DockerID) error
```

Comment:
I created the interface in the 'host' if someone wants to use it.
Did not want to call that flat search function everytime, especially when there is no candidate consumer today.

Taken care of all the rest of the feedback. Thanks.